zcash_keys/src/keys.rs

```diff
@@ -678,42 +708,69 @@ impl UnifiedFullViewingKey {
             sapling,
             #[cfg(feature = "orchard")]
             orchard,
-            unknown,
+            unknown_data,
+            expiry_height,
+            expiry_time,
+            unknown_metadata,
         })
```

Comment:
This does not check the requirement on the allowed receivers and metadata. It may be that an invalid combination can't occur here because of checks in `unified::Ufvk::decode(encoding)`, but if so it's not obvious.

Reply:
673e1a21d28209103dce50ede358491e5978d007